### DISCUSSION TITLE
Is this react hook (also for Next.js) correct for realtime?

### DISCUSSION THREAD

**Discussion by ChrisMGeo** [pocketbase/pocketbase#4256] [Q&A]
Hello I'm making a next.js with typescript app using pocketbase, and was wondering how realtime worked. The code below is what I have rn. Before this version, instead of using `switch (recordSubscription.action)` I just did `getFullList` again.
```typescript
import { db } from "@/consts/pocketbase";
import { type RecordFullListOptions, type SendOptions, RecordModel } from "pocketbase";
import { useState, useEffect } from "react";

const useGetFullList = <T extends RecordModel>(idOrName: string, options?: RecordFullListOptions | undefined, sendOptions?: SendOptions | undefined) => {
  const [records, setRecords] = useState<T[]>([]);
  useEffect(() => {
    db.collection(idOrName).getFullList<T>(options).then(records => setRecords(records));
  }, []);
  useEffect(() => {
    db.collection(idOrName).subscribe('*', (recordSubscription) => {
      switch (recordSubscription.action) {
        case "create":
          const newRecord = recordSubscription.record as T;
          setRecords(records => [...records, newRecord]);
          break;
        case "update":
          const updatedRecord = recordSubscription.record as T;
          setRecords(records => records.map(record => record.id === updatedRecord.id ? updatedRecord : record));
          break;
        case "delete":
          const deletedRecord = recordSubscription.record as T;
          setRecords(records => records.filter(record => record.id !== deletedRecord.id));
          break;
      }
    }, sendOptions);
    return () => {
      db.collection(idOrName).unsubscribe('*');
    };
  }, []);
  return records;
};
export default useGetFullList;
```

However both have one issue, being that when used in Next.js it gives me an auto-cancellation error, as well as long waiting time before initially fetching the data:
```
Unhandled Runtime Error
ClientResponseError 0: The request was autocancelled. You can find more info in https://github.com/pocketbase/js-sdk#auto-cancellation.

Call Stack
eval
node_modules\pocketbase\dist\pocketbase.es.mjs (1:31878)
```
Any tips on how to prevent this?
EDIT: Also I can't seem to find a `PocketBase` type for db, so I import it instead. If anyone knows the right name please do tell.

**Comment by ibanks42** [comment]
You can call `db.autoCancellation(false);`

**Comment by acortiv** [comment]
You need to create a new instance of pocketbase for each client request I've found (apparently this is best practice; for middleware and server components set the pb auth model token using cookies).  Exporting the const works for server components, but for client components, first create a method that initializes a new pocketbase instance, then call that method within an exported method for CRUD, then import that exported CRUD method into your client component. Let me know if you have any questions and I can walk you through it.

**Reply by bisvillallai** [reply]
Mate I'm super confused, can you please elaborate a bit more with this? 
I'm trying to have a realtime toast in the frontend when a record is updated in certain collection, so currently my code is:

function notify() {
  console.log('Notified')
  Notification('There was an update')
}

global.EventSource = EventSource

const pb = new PocketBase('https://rest.of.url') as TypedPocketBase
pb.autoCancellation(false)

pb.collection('name').subscribe('*', () => {
  notify()
})

I get the text "Notified" on the server console, but for the client I get this error:
 ⨯ lib\pocketbase.ts (8:3) @ Notification
 ⨯ uncaughtException: TypeError: (0 , sonner__WEBPACK_IMPORTED_MODULE_1__.toast) is not a function

"Notification" is a client component with just a regular toast from sonner.
Thank you in advance

**Reply by TheHypnoo** [reply]
Just so you understand, the server cannot have a singleton. On the client, it is possible to have a singleton.
